src/Microsoft.Health.ControlPlane.Core/Features/Persistence/IControlPlaneDataStore.cs

```diff
@@ -19,5 +19,13 @@ public interface IControlPlaneDataStore
         Task<UpsertResponse<IdentityProvider>> UpsertIdentityProviderAsync(IdentityProvider identityProvider, string eTag, CancellationToken cancellationToken);
 
         Task DeleteIdentityProviderAsync(string name, string eTag, CancellationToken cancellationToken);
+
+        Task<Role> GetRoleAsync(string name, CancellationToken cancellationToken);
+
+        Task<UpsertResponse<Role>> UpsertRoleAsync(Role role, string eTag, CancellationToken cancellationToken);
+
+        Task<IEnumerable<Role>> GetRoleAllAsync(CancellationToken cancellationToken);
```

Comment:
Let's line up the naming with the `GetAllIdentityProvidersAsync` -- e.g. `GetAllRolesAsync`

Reply:
Resolved